python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/mpls.j2

```diff
@@ -35,3 +35,138 @@ mpls icmp fragmentation-needed tunneling
 mpls icmp ttl-exceeded tunneling
 {%     endif %}
 {% endif %}
+{% if mpls.rsvp is arista.avd.defined %}
+!
+mpls rsvp
+{%     if mpls.rsvp.refresh is arista.avd.defined %}
+{%         if mpls.rsvp.refresh.interval is arista.avd.defined %}
+   refresh interval {{ mpls.rsvp.refresh.interval }}
+{%         endif %}
+{%         if mpls.rsvp.refresh.method is arista.avd.defined %}
+   refresh method {{ mpls.rsvp.refresh.method }}
+{%         endif %}
+{%     endif %}
+{%     if mpls.rsvp.hello.interval is arista.avd.defined and mpls.rsvp.hello.multiplier is arista.avd.defined %}
+   hello interval {{ mpls.rsvp.hello.interval }} multiplier {{ mpls.rsvp.hello.multiplier }}
+{%     endif %}
+{%     if mpls.rsvp.authentication is arista.avd.defined %}
+{%         set auth = mpls.rsvp.authentication %}
+{%         if auth.type is arista.avd.defined %}
+   authentication type {{ auth.type }}
+{%         endif %}
+{%         if auth.sequence_number_window is arista.avd.defined %}
+   authentication sequence-number window {{ auth.sequence_number_window }}
+{%         endif %}
+{%         for password_index in auth.password_indexes | arista.avd.natural_sort('index') %}
+{%             if password_index.password is arista.avd.defined %}
+{%                 if password_index.password_type is arista.avd.defined %}
+   authentication index {{ password_index.index }} password {{ password_index.password_type }} {{ password_index.password | arista.avd.hide_passwords(hide_passwords) }}
+{%                 else %}
+   authentication index {{ password_index.index }} password {{ password_index.password | arista.avd.hide_passwords(hide_passwords) }}
+{%                 endif %}
+{%             endif %}
+{%         endfor %}
+{%         if auth.active_index is arista.avd.defined %}
+   authentication index {{ auth.active_index }} active
+{%         endif %}
+{%     endif %}
+{%     if mpls.rsvp.neighbors is arista.avd.defined %}
+{%         set with_neighbor_ipv4_address = mpls.rsvp.neighbors | selectattr('ip_address', 'arista.avd.defined') | arista.avd.natural_sort('ip_address') %}
+{%         set with_neighbor_ipv6_address = mpls.rsvp.neighbors | selectattr('ipv6_address', 'arista.avd.defined') | arista.avd.natural_sort('ipv6_address') %}
+{%         set sorted_ip_addresses =  with_neighbor_ipv4_address | list + with_neighbor_ipv6_address | list %}
+{%         for neighbor in sorted_ip_addresses | arista.avd.default([]) %}
+{%             if neighbor.ip_address is arista.avd.defined %}
+{%                 set ip_address = neighbor.ip_address %}
+{%             else %}
```

Comment:
```suggestion
{%             elif neighbor.ipv6_address is arista.avd.defined %}
```

Reply:
as we are already forming two separate list ipv4 and ipv6 and joining them so if ip_address is not there then it will always  ipv6_address. Even elif also ok but it block the coverage here as it need the case where both ip_address and ipv6_address is not defined.